.github/workflows/main.yml

```diff
@@ -0,0 +1,153 @@
+          role-session-name: GitHub_to_AWS_via_FederatedOIDC
+      - name: Run Meta Ntf using Centerpoint Model
+        uses: aws-actions/aws-codebuild-run-build@v1
+        env:
+            TEST_IMAGE: "meta"
+            TEST_MODEL: "centerpoint"
+        with:
+          env-vars-for-codebuild: |
+            TEST_IMAGE,
+            TEST_MODEL
+          project-name: CodeBuild-TestGithubOSML
+          buildspec-override: .github/codebuild/test.yml
+  RunLargeTifFloodTest:
+    needs: BuildOSML
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: us-west-2
+          role-to-assume: arn:aws:iam::${{ secrets.OSML_dev }}:role/GithubAction-AssumeRoleWithAction
+          role-session-name: GitHub_to_AWS_via_FederatedOIDC
+      - name: Run Large Tif using Flood Model
+        uses: aws-actions/aws-codebuild-run-build@v1
+        env:
+            TEST_IMAGE: "large"
+            TEST_MODEL: "flood"
+        with:
+          env-vars-for-codebuild: |
+            TEST_IMAGE,
+            TEST_MODEL
+          project-name: CodeBuild-TestGithubOSML
+          buildspec-override: .github/codebuild/test.yml
+  RunTileNtfAircraftTest:
+    needs: BuildOSML
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: us-west-2
+          role-to-assume: arn:aws:iam::${{ secrets.OSML_dev }}:role/GithubAction-AssumeRoleWithAction
+          role-session-name: GitHub_to_AWS_via_FederatedOIDC
+      - name: Run Tile Ntf using Aircraft Model
+        uses: aws-actions/aws-codebuild-run-build@v1
+        env:
+            TEST_IMAGE: "tile_ntf"
+            TEST_MODEL: "aircraft"
+        with:
+          env-vars-for-codebuild: |
+            TEST_IMAGE,
+            TEST_MODEL
+          project-name: CodeBuild-TestGithubOSML
+          buildspec-override: .github/codebuild/test.yml
+  RunTileTifAircraftTest:
+    needs: BuildOSML
+    runs-on: ubuntu-latest
+    steps:
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-region: us-west-2
+          role-to-assume: arn:aws:iam::${{ secrets.OSML_dev }}:role/GithubAction-AssumeRoleWithAction
+          role-session-name: GitHub_to_AWS_via_FederatedOIDC
+      - name: Run Tile Tif using Aircraft Model
+        uses: aws-actions/aws-codebuild-run-build@v1
+        env:
+            TEST_IMAGE: "tile_tif"
+            TEST_MODEL: "aircraft"
+        with:
+          env-vars-for-codebuild: |
+            TEST_IMAGE,
+            TEST_MODEL
+          project-name: CodeBuild-TestGithubOSML
+          buildspec-override: .github/codebuild/test.yml
+  DestroyOSML:
+    needs: [RunSmallTifCenterpointTest, RunMetaNtfCenterpointTest, RunLargeTifFloodTest, RunTileNtfAircraftTest, RunTileTifAircraftTest]
```

Comment:
Going add `if: ${{ always() }}` if any of the deployment/test fails. We need to ensure that we can safely delete all the stacks and resources.

Reply:
Need to add `BuildOSML` to `needs:`